rsconnect/api.py

```diff
@@ -268,6 +268,312 @@ def output_task_log(task_status, last_status, log_callback):
         return new_last_status
 
 
+class RSConnectExecutor:
+    def __init__(self, *args, **kwargs) -> None:
+        print(kwargs)
+        self.reset()
+        self._d = kwargs
```

Comment:
In general, this class should try to avoid pulling values from a dictionary but instead have its own named/typed attributes, again for tooling support. That would enable typechecking and prevent bugs that can creep in through dict key typos.